tests/drivers/pwm/pwm_gpio_loopback/dts/bindings/test-pwm-gpio-loopback.yaml

```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2024 Espressif Systems (Shanghai) Co., Ltd.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: |
+    This binding provides resources required to build and run the test
+    tests/drivers/pwm/pwm_gpio_loopback.
+
+compatible: "test-pwm-gpio-loopback"
+
+properties:
+  in-gpios:
+    type: phandle-array
+    required: true
+    description: |
+      GPIO to be used as input to check and validate PWM timing.
```

Comment:
I know the existing test uses a bindings file for the properties, but this is actually wrong since `zephyr,user` has been added. custom properties used for samples and tests, which don't reflect real hardware, as compatibles do, should use the `zephyr,user` node:
```
zephyr,user {
        in-gpios = <>;
};
```
and specific instances are accessed using aliases:
```
aliases {
        pwm = &pwm0;
};
```
See https://github.com/zephyrproject-rtos/zephyr/blob/d789f339efb78a136a10c74efdeeb963e085e104/tests/drivers/comparator/gpio_loopback/boards/nrf5340dk_nrf5340_cpuapp.overlay#L13-L21 and https://github.com/zephyrproject-rtos/zephyr/blob/d789f339efb78a136a10c74efdeeb963e085e104/tests/drivers/comparator/gpio_loopback/src/test.c#L12-L13

This allows board specific overlays to map the aliases and user properties to actual devices, negating the need for
```
#if DT_NODE_HAS_STATUS_OKAY(DT_ALIAS(pwm_0))
#define PWM_DEV_NODE DT_ALIAS(pwm_0)
#elif DT_NODE_HAS_STATUS_OKAY(DT_ALIAS(pwm_1))
#define PWM_DEV_NODE DT_ALIAS(pwm_1)
#elif DT_NODE_HAS_STATUS_OKAY(DT_ALIAS(pwm_2))
#define PWM_DEV_NODE DT_ALIAS(pwm_2)
#elif DT_NODE_HAS_STATUS_OKAY(DT_ALIAS(pwm_3))
#define PWM_DEV_NODE DT_ALIAS(pwm_3)
#else
#error "Please define a PWM device"
#endif
```
which simply becomes:
```
#define PWM_DEV_NODE DT_ALIAS(pwm)
```